studies/views/study_views.py

```diff
@@ -106,6 +106,15 @@ def form_valid(self, form):
         for study_type in form.fields["study_types"].choices:
             form_value = study_type[0] in form.data.getlist("study_types")
             form.instance.__setattr__(study_type[0], form_value)
+        if not form.instance.has_intervention and hasattr(
+            form.instance, "intervention"
+        ):
+            form.instance.intervention.delete()
+        if not form.instance.has_observation and hasattr(form.instance, "observation"):
+            form.instance.observation.delete()
+        if not form.instance.has_sessions and form.instance.session_set.all():
+            for session in form.instance.session_set.all():
+                session.delete()
```

Comment:
It can be annoying to have these instances dangling around when the type is no longer relevant. But the unintentional feature that comes with it is that users can undo deletion by re-enabling it. Which is nice and we would probably want to keep.

Best to solve the issues with these objects appearing elsewhere symptomatically.

To do this right, we could move to a `Study.get_sessions()` function where needed. That function would return no sessions if not `study.as_sessions`.

Or we could assign custom managers to things like `study_set` and `sessions_set` that do the same.

Just a suggestion. A couple if statements to fix the diffs is fine by me as well, just a little harder to maintain.